lib/NetworkManager.js

```diff
@@ -268,6 +268,8 @@ class NetworkManager extends EventEmitter {
     if (!request)
       return;
     request._completePromiseFulfill.call(null);
+    // save loading finished timestamp and not loading started timestamp
+    request._timestamp = event.timestamp;
```

Comment:
shall we have `request.timestamp()` as the start time, and `response.timestamp()` as the loading-finished time? This seems more logical to me.

Reply:
I partly understand but how do we keep the timestamps that a request failed or finished?

.timestampFinished()
.timestampFailed()

on Request object? So when requests fail I still know the performance hit it got?

Reply:
> I partly understand but how do we keep the timestamps that a request failed or finished?

Good point. I like your suggestions; I'd prefix the time with `"monotonic"` so that it's less confusing.

```js
request.monotonicTimeStart() // when the request started
request.monotonicTimeEnd() // when the request failed / finished
```

cc @paulirish for API suggestions

Reply:
I will wait a day or 2 on @paulirish before changing the code. I do not want to add extra metadata, but `request.hasFailed()` could be handy? Or just stick with `.monotonicxxx`?

Reply:
Request failed state is dispatched with events; do you want the `request.hasFailed()` for convenience?

Reply:
@aslushnikov perhaps it is confusing for developers, but it was just a logical thought to show no `response` or `response.timestamp` can be expected if that request has failed.